identity/README.md

```diff
@@ -0,0 +1,299 @@
+# Identity Service
+
+Author(s): Niels de Vos
+
+This document and the Identity Service is based on the [Container Storage
+Interface specification][csi_spec].
+
+## Terminology
+
+| Term       | Definition                                                                               |
+| ---------- | ---------------------------------------------------------------------------------------- |
+| CSI        | [Container Storage Interface][csi_spec].                                                 |
+| CSI-driver | A component that implements CSI service RPCs, the CSI Controller or NodePlugin.          |
+| CO         | Container Orchestration system that communicates with users and CSI-Addons service RPCs. |
+| SP         | Storage Provider, the vendor of a CSI-driver implementation.                             |
+| RPC        | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).            |
+
+## Objective
+
+Define an API that makes it possible for the CSI-Addons Controller(s) to
+communicate with the CSI-Addons side-cars that are running with the CSI-driver
+components.
+
+### Goals in MVP
+
+The API will
+
+* make it possible for worker nodes (CSI-Addons side-car) to announce
+  themselves
+* have the CSI-Addons Controller list all announced CSI-Addons side-cars and
+  CSI-drivers
+* provide details about CSI-Addons features the CSI-driver supports
+
+### Non-Goals in MVP
+
+This will be limited to node discovery functionality only. CSI-Addons features
+will be documented in their own specifications. The available capabilities will
+get extended when new CSI-Addons operations are proposed.
+
+## Solution Overview
+
+CSI-Addons side-cars for the CSI-Controller and CSI-NodePlugins will announce
+themselves at the CO.
+
+The CSI-Addons Controller reacts on events generated by the CO (e.g. Kubernetes
+CRDs) that trigger actions. When a CSI-Addons Node is announced to the CO, the
+CSI-Addons Controller will connect to the Node and request the available
+capabilities.
```

Comment:
This looks more like a kubernetes implementation not related to csi-addons.

Reply:
I'll rephrase things a bit, and rename `CSI-Addons Controller` to `CSI-Addons CO plugin` to make is sound less Kubernetes-y.